### PR TITLE
geos: 3.10.2 -> 3.11.0

### PR DESCRIPTION
**Pull request by willcohen**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
On macos 10.15:

```
builder for '/nix/store/3l5wjhd8rygd565hlgl5bwaympkwcgdy-python3.9-pygeos-0.12.0.drv' failed with exit code 1; last 10 log lines:
  -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
  =========================== short test summary info ============================
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry0]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry3]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry4]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry5]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry6]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry7]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry8]
  = 7 failed, 2614 passed, 22 skipped, 1 xfailed, 22 xpassed, 5 warnings in 5.63s =
```

```
builder for '/nix/store/7kg4maa70cahiyw8nr9njhd2jvr3rcr1-python3.9-Shapely-1.8.2.drv' failed with exit code 1; last 10 log lines:
      raise GeometryTypeError("Splitting a Polygon with a %s is not supported" % splitter.type)
  
  tests/test_substring.py::SubstringTestCase::test_raise_type_error
    /nix/store/m1lyy5x531vyi8sxiy3zjabddva72f5q-python3.9-Shapely-1.8.2/lib/python3.9/site-packages/shapely/ops.py:639: ShapelyDeprecationWarning: GeometryTypeError will derive from ShapelyError and not TypeError or ValueError in Shapely 2.0.
      raise GeometryTypeError("Can only calculate a substring of LineString geometries. A %s was provided." % geom.type)
  
  -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
  =========================== short test summary info ============================
  FAILED tests/test_parallel_offset.py::OperationsTestCase::test_parallel_offset_linestring
  ===== 1 failed, 463 passed, 18 deselected, 2 xfailed, 27 warnings in 1.97s =====
```

Same for the python 3.10 versions of those packages

**Comment by willcohen** [comment]
Including @knedlsepp for questions about disabling shapely tests.

**Comment by risicle** [comment]
A failing test suite is an indication of broken functionality unless we know otherwise. Disabling tests does not solve this.

**Review by risicle — CHANGES_REQUESTED** [review]
Disabling test suites is not a solution to failing tests.

File: pkgs/development/python-modules/pygeos/default.nix [hunk 1]
```diff
@@ -17,6 +17,12 @@ buildPythonPackage rec {
     sha256 = "sha256-PEFULvZ8ZgFfRDrj5uaDUDqKIh+cJPsjgPauQq7RYAo=";
   };
 
+  patches = [
+    # Adapt https://github.com/shapely/shapely/commit/4889bd2d72ff500e51ba70d5b954241878349562,
+    # backporting to pygeos
+    ./fix-for-geos-3-11.patch
```
Comment (risicle) [inline pkgs/development/python-modules/pygeos/default.nix:23]: If it's just a matter of extracting a single file's diff from the commit, you can just do

```
(fetchpatch {
  name = "fix-tests-geos-3.11.patch";
  url = "https://github.com/shapely/shapely/commit/4889bd2d72ff500e51ba70d5b954241878349562.patch";
  includes = [ "shapely/tests/test_constructive.py" ];
  sha256 = "...";
})
```

**Comment by willcohen** [comment]
As with the 3.10 bump, both my (not so fast) my x86 Mac and my (little bit faster) Linux box are both hanging on their nixpkgs-review so I can’t totally confirm that the full rebuild works, but they’ve both at least gotten past the pygeos and shapely packages correctly.

**Comment by kirillrdy** [comment]
> As with the 3.10 bump, both my (not so fast) my x86 Mac and my (little bit faster) Linux box are both hanging on their nixpkgs-review so I can’t totally confirm that the full rebuild works, but they’ve both at least gotten past the pygeos and shapely packages correctly.

on my computer
```
7 packages marked as broken and skipped:
python310Packages.basemap python310Packages.descartes python310Packages.worldengine python39Packages.basemap python39Packages.descartes python39Packages.worldengine worldengine-cli

63 packages failed to build:
apacheHttpdPackages.mod_tile cloudcompare gdal gdal_2 gmt gplates grass libLAS mapcache mapnik mapproxy mapserver merkaartor openorienteering-mapper osmscout-server paraview pdal pdal_2_3 perl532Packages.Tirex perl534Packages.Tirex postgresql11Packages.postgis postgresql12Packages.postgis postgresql13Packages.postgis postgresqlPackages.postgis python310Packages.asf-search python310Packages.bsuite python310Packages.cartopy python310Packages.fiona python310Packages.geopandas python310Packages.mask-rcnn python310Packages.opensfm python310Packages.osmnx python310Packages.plotnine python310Packages.pygmt python310Packages.python-mapnik python310Packages.rasterio python310Packages.rlax python310Packages.wktutils python39Packages.asf-search python39Packages.bsuite python39Packages.cartopy python39Packages.fiona python39Packages.gdal python39Packages.geopandas python39Packages.mask-rcnn python39Packages.opensfm python39Packages.osmnx python39Packages.plotnine python39Packages.pygmt python39Packages.python-mapnik python39Packages.rasterio python39Packages.rlax python39Packages.wktutils pytrainer qgis qgis-ltr qlandkartegt qmapshack saga sumo t-rex udig valhalla

33 packages built:
cura flatcam geos haskellPackages.hgeos librttopo libspatialite moonraker python310Packages.aplpy python310Packages.flask-admin python310Packages.geoalchemy2 python310Packages.imgaug python310Packages.owslib python310Packages.preprocess-cancellation python310Packages.pygeos python310Packages.pyosmium python310Packages.pyproj python310Packages.shapely python310Packages.spectral-cube python310Packages.uranium python39Packages.aplpy python39Packages.flask-admin python39Packages.geoalchemy2 python39Packages.imgaug python39Packages.owslib python39Packages.preprocess-cancellation python39Packages.pygeos python39Packages.pyosmium python39Packages.pyproj python39Packages.shapely python39Packages.spectral-cube python39Packages.uranium readosm spatialite_tools
```

**Comment by kirillrdy** [comment]
I've tried building some of the failed packages manually and they do build, so maybe something with nixpkgs-review

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Comment by willcohen** [comment]
Thank you for running -- my linux box was still chewing on it with likely another day to go!

**Comment by risicle** [comment]
`pkgsMusl` and `pkgsCross.aarch64-multiplatform` variants also build happily.

**Comment by willcohen** [comment]
Thank you!

**Pull request merged**